ICAFIX/scripts/RecleanClassify.sh

```diff
@@ -0,0 +1,276 @@
+
+tempfiles_create MMPSubRegionROIs_temp_XXXXXX.dlabel.nii roisubtemp
+wb_command -cifti-create-dense-from-template "$GrayOrdinateTemplateFile" "$roisubtemp" -cifti "$SubRegionParcellation"
+
+tempfiles_create VisualROI_temp_XXXXXX.dlabel.nii VisualROItemp
+tempfiles_add "$VisualROItemp"_labels.dlabel.nii
+wb_command -cifti-label-import "$roitemp" "$HCPPIPEDIR"/global/config/Visual.ROI.txt "$VisualROItemp"_labels.dlabel.nii -discard-others
+wb_command -cifti-math 'x > 0' "$VisualROItemp" -var x "$VisualROItemp"_labels.dlabel.nii
+
+tempfiles_create LanguageROI_temp_XXXXXX.dlabel.nii LanguageROItemp
+tempfiles_add "$LanguageROItemp"_labels.dlabel.nii
+wb_command -cifti-label-import "$roitemp" "$HCPPIPEDIR"/global/config/Language.ROI.txt "$LanguageROItemp"_labels.dlabel.nii -discard-others
+wb_command -cifti-math 'x > 0' "$LanguageROItemp" -var x "$LanguageROItemp"_labels.dlabel.nii
+
+tempfiles_create SubRegionsROI_temp_XXXXXX.dscalar.nii SubRegionsROItemp
+tempfiles_add "$SubRegionsROItemp"_labels.dlabel.nii "$SubRegionsROItemp"_all_labels.dlabel.nii
+wb_command -cifti-label-import "$roisubtemp" "$HCPPIPEDIR"/global/config/SubRegions.ROI.txt "$SubRegionsROItemp"_labels.dlabel.nii -discard-others
+wb_command -cifti-all-labels-to-rois "$SubRegionsROItemp"_labels.dlabel.nii 1 "$SubRegionsROItemp"
+
+# shortcut in case the folder gets renamed
+this_script_dir=$(dirname "$0")
+#all arguments are strings, so we can can use the same argument list for compiled and interpreted
+matlab_argarray=("$StudyFolder" "$Subject" "$fMRIListName" "$subjectExpectedTimepoints" "$HighPass" "$fMRIResolution" "$CorticalParcellationFile" "$WMLabelFile" "$CSFLabelFile" "$VisualROItemp" "$LanguageROItemp" "$SubRegionsROItemp" "$NonGreyParcelsFile")
+
+case "$MatlabMode" in
+    (0)
+        matlab_cmd=("$this_script_dir/Compiled_computeRecleanFeatures/run_computeRecleanFeatures.sh" "$MATLAB_COMPILER_RUNTIME" "${matlab_argarray[@]}")
+        log_Msg "running compiled matlab command: ${matlab_cmd[*]}"
+        "${matlab_cmd[@]}"
+        ;;
+    (1 | 2)
+        #reformat argument array so matlab sees them as strings
+        matlab_args=""
+        for thisarg in "${matlab_argarray[@]}"
+        do
+            if [[ "$matlab_args" != "" ]]
+            then
+                matlab_args+=", "
+            fi
+            matlab_args+="'$thisarg'"
+        done
+        
+        matlabcode="
+            addpath('$HCPPIPEDIR/global/matlab');
+            addpath('$HCPPIPEDIR/global/scripts');
+            addpath('$HCPCIFTIRWDIR');
+            addpath('$this_script_dir');
+            computeRecleanFeatures($matlab_args);"
+
+        log_Msg "running matlab code: $matlabcode"
+        "${matlab_interpreter[@]}" <<<"$matlabcode"
+        echo
+        ;;
+esac
+
+# hardcoded
+FixProbThresh="10"
+
+# inference each subjects under python environment
+for fMRIName in ${fMRINamesToUse} ; do
```

Comment:
```suggestion
for fMRIName in "${fMRINamesToUse[@]}" ; do
```
Otherwise it will only do the first in the array.